components/status-message/spec/StatusMessage.ts

```diff
@@ -0,0 +1,57 @@
+import { createElement as h } from 'react';
+import { mount } from '@not-govuk/component-test-helpers';
+import StatusMessage from '../src/StatusMessage';
+import {A} from "@not-govuk/components";
+
+describe('StatusMessage', () => {
+  describe('when given a status', () => {
+    const component = mount(h(StatusMessage, { status: 'My heading' }));
+    it('should show the status', () => expect(component.text()).toMatch(/My heading/));
+
+    describe('and extra contents', () => {
+      const component = mount(h(StatusMessage, { status: 'My heading' }, 'extra info'));
+      it('should show the status and the extra contents', () =>
+        expect(component.text()).toMatch(/My heading\s+-\s+extra info/)
+      );
+
+      describe('with an action', () => {
+        const component = mount(h(StatusMessage, {
+          status: 'My heading',
+          actions: h(A, { href: '#' }, 'Run')
+        }, 'extra info'));
+        it('should show the status, extra contents, and action', () =>
+          expect(component.text()).toMatch(/My heading\s+-\s+extra info\s*Run/)
+        );
```

Comment:
I wonder if it's worth breaking this up into 3 tests and using `toContain` (or whatever it's called).

Reply:
Refactored to split into more targeted checks